ibm/resource_ibm_satellite_cluster.go

```diff
@@ -0,0 +1,766 @@
+	isClusterDeleteDone    = "done"
+	isWorkderDeployed      = "deployed"
+)
+
+func resourceIBMSatelliteCluster() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMSatelliteClusterCreate,
+		Read:     resourceIBMSatelliteClusterRead,
+		Update:   resourceIBMSatelliteClusterUpdate,
+		Delete:   resourceIBMSatelliteClusterDelete,
+		Importer: &schema.ResourceImporter{},
+
+		CustomizeDiff: customdiff.Sequence(
+			func(_ context.Context, diff *schema.ResourceDiff, v interface{}) error {
+				return immutableResourceCustomizeDiff([]string{"name", "location", "resource_group_id"}, diff)
+			},
+			func(_ context.Context, diff *schema.ResourceDiff, v interface{}) error {
+				return resourceTagsCustomizeDiff(diff)
+			},
+		),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Read:   schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+			Update: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The unique name for the new IBM Cloud Satellite cluster",
+			},
+			"location": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name or ID of the Satellite location",
+			},
+			"state": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The lifecycle state of the cluster.",
+			},
+			"kube_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+				DiffSuppressFunc: func(k, o, n string, d *schema.ResourceData) bool {
+					if o == "" {
+						return false
+					}
+					new := strings.Split(n, ".")
+					old := strings.Split(o, ".")
+
+					if strings.Compare(new[0]+"."+strings.Split(new[1], "_")[0], old[0]+"."+strings.Split(old[1], "_")[0]) == 0 {
+						return true
+					}
+					return false
+				},
+				Description: "The OpenShift Container Platform version",
+			},
+			"wait_for_worker_update": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Wait for worker node to update during kube version update.",
+			},
+			"patch_version": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Kubernetes patch version",
+			},
+			"retry_patch_version": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "Argument which helps to retry the patch version updates on worker nodes. Increment the value to retry the patch updates if the previous apply fails",
+			},
+			"master_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"enable_config_admin": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Optional:    true,
+				Description: "Grant cluster admin access to Satellite Config to manage Kubernetes resources.",
+			},
+			"workers": {
```

Comment:
change to worker_count to have similar naming convention across all resources/datasources

Reply:
Changed to worker_count